x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -575,40 +572,38 @@ public void getAutodetectParams(Job job, Consumer<AutodetectParams> consumer, Co
-                                    } else {
-                                        SearchHits hits = searchResponse.getHits();
-                                        long hitsCount = hits.getHits().length;
-                                        if (hitsCount == 0) {
-                                            SearchRequest searchRequest = msearch.request().requests().get(i);
-                                            LOGGER.debug("Found 0 hits for [{}]", new Object[]{searchRequest.indices()});
-                                        } else {
-                                            for (SearchHit hit : hits) {
-                                                try {
-                                                    parseAutodetectParamSearchHit(jobId, paramsBuilder, hit);
-                                                } catch (Exception e) {
-                                                    errorHandler.accept(e);
-                                                    return;
-                                                }
-                                            }
+                                }
+                                SearchResponse searchResponse = itemResponse.getResponse();
+                                ShardSearchFailure[] shardFailures = searchResponse.getShardFailures();
+                                int unavailableShards = searchResponse.getTotalShards() - searchResponse.getSuccessfulShards();
+                                if (shardFailures != null && shardFailures.length > 0) {
+                                    LOGGER.error("[{}] Search request returned shard failures: {}", jobId,
+                                            Arrays.toString(shardFailures));
+                                    errorHandler.accept(new ElasticsearchException(
+                                            ExceptionsHelper.shardFailuresToErrorMsg(jobId, shardFailures)));
+                                    return;
+                                }
+                                if (unavailableShards > 0) {
+                                    errorHandler.accept(new ElasticsearchException("[" + jobId
+                                            + "] Search request encountered [" + unavailableShards + "] unavailable shards"));
+                                    return;
+                                }
+                                SearchHits hits = searchResponse.getHits();
+                                long hitsCount = hits.getHits().length;
+                                if (hitsCount == 0) {
+                                    SearchRequest searchRequest = msearch.request().requests().get(i);
+                                    LOGGER.debug("Found 0 hits for [{}]", new Object[]{searchRequest.indices()});
+                                    return;
+                                }
+                                    for (SearchHit hit : hits) {
```

Comment:
Is there an redundant indentation here or is it github messing with me?

Reply:
fixed :)